### PR TITLE
feat: store test cluster logs and improve test readme [DET-3269]

### PR DESCRIPTION
**Pull request by hamidzr**
## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/tests/README.md [hunk 1]
```diff
@@ -37,8 +37,30 @@ Before the tests can be started we need to build the cluster including the WebUI
...
+### Debugging Test Issues
+
+For reproducing and catching test flakes there is a simple helper script `./bin/try-for-flake.sh`.
+Just executing the script will run `make test` over and over until it hits a error.
+
+To speed up this process you can:
+
+1. try to avoid the whole cluster set up and tear down on each iteration
+2. limit the test scope:
+  - use `.skip` on each unwanted test suite or `.only` on the target suite.
```
Comment (hkang1) [inline webui/tests/README.md:60]: question: does this actually work with our cypress setup? It looks like it's a mix bag. I tried applying `.only` to the very first test in `02-loginout.spec.ts` and it ran all the other test files fully, and when it came to `02-loginout.spec.ts` it did run just the first test where I applied `.only`... Is that the correct behavior or is there a weird setup going on?

Comment (hkang1) [reply webui/tests/README.md:60]: update: here's what worked for blocking full test files. If you apply `x` to the top level `describe` (`xdescribe`) it stops the entire `describe` block in that file from running. So to isolate the tests I want to run, I need to add `x` to all the other files.

File: webui/tests/bin/e2e-tests.py [hunk 2]
```diff
@@ -63,13 +63,15 @@ def teardown_cluster(config):
 @contextmanager
 def det_cluster(config):
     try:
-        yield setup_cluster(config)
+        with open(str(results_dir.joinpath("cluster.stdout.logs")), "w") as f:
```
[the remark this answers is not in the document]
Comment (hamidzr) [reply webui/tests/bin/e2e-tests.py]: changed up this section

File: webui/tests/bin/e2e-tests.py [hunk 3]
```diff
@@ -31,8 +31,8 @@ def run(cmd: List[str], config) -> None:
     return subprocess.check_call(cmd, env=config["env"])
```
[the remark this answers is not in the document]
Comment (hamidzr) [reply webui/tests/bin/e2e-tests.py:32]: `run` was missing since `yarn` didn't need it

[the remark this answers is not in the document]
Comment (hamidzr) [reply webui/tests/bin/e2e-tests.py:32]: good catch!

[the remark this answers is not in the document]
Comment (hamidzr) [reply webui/tests/bin/e2e-tests.py:32]: did this happen every time? doesn't happen for me 

Screen Capture_select-area_20200608160150

File: webui/tests/README.md [hunk 4]
```diff
@@ -37,8 +37,30 @@ Before the tests can be started we need to build the cluster including the WebUI
...
+
+## Test Development
+
+Use `make dev-tests` to set up for test development and then proceed to add new
+tests suites or update and rebuild the WebUI artifacts to see changes in tests.
+
+### Debugging Test Issues
+
+For reproducing and catching test flakes there is a simple helper script `./bin/try-for-flake.sh`.
+Just executing the script will run `make test` over and over until it hits a error.
```
[the remark this answers is not in the document]
Comment (hamidzr) [reply webui/tests/README.md:54]: the script doesn't support this option we can add that (separately) though.

File: webui/tests/README.md (hunk 1 above)
Comment (hamidzr) [reply webui/tests/README.md:60]: it was mixed bag last time I used it. from my googling, it's about the Jest version that Cypress is using. It is supposed to work so I think it's safe to have it in the readme. I usually just delete the unwanted test suites.
https://github.com/cypress-io/cypress/issues/5441
https://github.com/cypress-io/cypress/issues/632
https://github.com/cypress-io/cypress/issues/5345

File: webui/tests/README.md [hunk 5]
```diff
@@ -37,8 +37,30 @@ Before the tests can be started we need to build the cluster including the WebUI
...
+
+To speed up this process you can:
+
+1. try to avoid the whole cluster set up and tear down on each iteration
```
[the remark this answers is not in the document]
Comment (hamidzr) [reply webui/tests/README.md:58]: it depends on if the target tests are re-runnable or not. I added more description here. let me know if that helps

File: webui/tests/bin/e2e-tests.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/tests/bin/e2e-tests.py:32]: I get the error below now when I try `./bin/try-for-flake.sh`. It looks like it's an issue around the `det` CLI with changing password.
```
❯ ./bin/try-for-flake.sh 5
./bin/try-for-flake.sh: line 8: 5: command not found
run #1
python ./bin/e2e-tests.py e2e-tests
+ rm -r /Users/calebkang/Projects/determined/webui/tests/results
+ mkdir -p /Users/calebkang/Projects/determined/webui/tests/results
setting up the cluster..
+ make -C /Users/calebkang/Projects/determined/webui/tests/test-cluster start-db
docker-compose down
Removing network test-cluster_default
WARNING: Network test-cluster_default not found.
docker-compose up -d
Creating network "test-cluster_default" with the default driver
Creating test-cluster_determined-db_1 ... done
cluster pid: 20493
+ rm -r /Users/calebkang/Projects/determined/webui/tests/results
+ mkdir -p /Users/calebkang/Projects/determined/webui/tests/results
+ python /Users/calebkang/Projects/determined/webui/tests/bin/createUserAndExperiments.py
setting up users..
Traceback (most recent call last):
  File "/Users/calebkang/Projects/determined/webui/tests/bin/createUserAndExperiments.py", line 58, in <module>
    create_user(USER_WITH_PASSWORD_USERNAME, USER_WITH_PASSWORD_PASSWORD)
  File "/Users/calebkang/Projects/determined/webui/tests/bin/createUserAndExperiments.py", line 43, in create_user
    p.expect("New password.*:")
  File "/Users/calebkang/.local/share/virtualenvs/determined-42Ca9GpV/lib/python3.6/site-packages/pexpect/spawnbase.py", line 344, in expect
    timeout, searchwindowsize, async_)
  File "/Users/calebkang/.local/share/virtualenvs/determined-42Ca9GpV/lib/python3.6/site-packages/pexpect/spawnbase.py", line 372, in expect_list
    return exp.expect_loop(timeout)
  File "/Users/calebkang/.local/share/virtualenvs/determined-42Ca9GpV/lib/python3.6/site-packages/pexpect/expect.py", line 179, in expect_loop
    return self.eof(e)
  File "/Users/calebkang/.local/share/virtualenvs/determined-42Ca9GpV/lib/python3.6/site-packages/pexpect/expect.py", line 122, in eof
    raise exc
pexpect.exceptions.EOF: End Of File (EOF). Empty string style platform.
<pexpect.pty_spawn.spawn object at 0x10ab15278>
command: /Users/calebkang/.local/share/virtualenvs/determined-42Ca9GpV/bin/det
args: ['/Users/calebkang/.local/share/virtualenvs/determined-42Ca9GpV/bin/det', '-u', 'admin', 'user', 'change-password', 'user-w-pw']
buffer (last 100 chars): b''
before (last 100 chars): b"bject at 0x106a4f0f0>: Failed to establish a new connection: [Errno 61] Connection refused',))\r\n\x1b[0m"
after: <class 'pexpect.exceptions.EOF'>
match: None
match_index: None
exitstatus: None
flag_eof: True
pid: 20633
child_fd: 6
closed: False
timeout: 30
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile(b'New password.*:')
```

[the remark this answers is not in the document]
Comment (hkang1) [reply webui/tests/bin/e2e-tests.py:32]: after rebasing to the tip of the master this runs great

**Pull request merged**